python/ray/experimental/shuffle.py

```diff
@@ -308,6 +310,12 @@ def output_writer_non_streaming(
     )
     delta = time.time() - start
 
+    ray.get(monitor_actor.stop_run.remote())
```

Comment:
Added this to better monitor memory usage of this test since it has some OOM issue before